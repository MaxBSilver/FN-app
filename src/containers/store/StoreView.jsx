import React, { Component } from "react";
import { getCurrentStoreItemsThunk } from "../../api/store";
import moment from "moment";
import { connect } from "react-redux";
import StoreItem from "../../components/storeItem/StoreItem";
import Loading from "../../components/loading/Loading";
export class StoreView extends Component {
  async componentDidMount() {
    if (this.props.storeItems.length === 0)
      await this.props.getCurrentStoreItemsThunk();
  }
  render() {
    const storeItems = this.props.storeItems.map(storeItem => (
      <StoreItem key={storeItem.itemId} {...storeItem} />
    ));
    return this.props.isLoading ? (
      <div className="store-container">
        <Loading />
      </div>
    ) : (
      <div className="store-container">
        <section className="store-info-container">
          <h2>
            Store<span>{moment().format("MMM Do YY")}</span>
          </h2>
          <p>
            The current item shop rotation for Fortnite Battle Royale - updates
            daily at <span>00:00 UTC</span> .
          </p>
        </section>
        <section className="store-item-container">{storeItems}</section>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  storeItems: state.storeItems,
  isLoading: state.isLoading,
  error: state.error
});

export const mapDispatchToProps = dispatch => ({
  getCurrentStoreItemsThunk: () => dispatch(getCurrentStoreItemsThunk())
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StoreView);
