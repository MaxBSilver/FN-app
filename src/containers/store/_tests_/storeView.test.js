import { mapStateToProps, mapDispatchToProps, StoreView } from "../StoreView";
import React from "react";

import { shallow } from "enzyme";
describe("StoreView", () => {
  describe("StoreView Component", () => {
    let wrapper;
    beforeEach(() => {
      const mockProps = { storeItems: [{ itemId: 1 }] };
      wrapper = shallow(<StoreView {...mockProps} />);
    });
    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe("mapStateToProps", () => {
    it("should return an object with the storeItems array", () => {
      const mockState = {
        storeItems: [{ title: "New Skins" }, { title: "New Items" }]
      };
      const expected = {
        storeItems: [{ title: "New Skins" }, { title: "New Items" }]
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
    it("should return an boolean", () => {
      const mockState = {
        isLoading: false
      };
      const expected = {
        isLoading: false
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
    it("should return an error message", () => {
      const mockState = {
        error: "Has Errored!"
      };
      const expected = {
        error: "Has Errored!"
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});
