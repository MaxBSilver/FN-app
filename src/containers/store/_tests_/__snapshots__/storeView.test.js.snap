// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreView StoreView Component should match snapshot 1`] = `ShallowWrapper {}`;
